src/main/java/org/xstefank/model/Utils.java

```diff
@@ -27,31 +29,30 @@
     public static final String MESSAGE = "message";
     public static final String URL = "url";
 
-    public static String readTokenFromProperties(String dirName, String fileName) {
-        InputStream is = null;
-        File dir = new File(dirName);
-        File fileProp = new File(dir, fileName);
-
-        try {
-            is = new FileInputStream(fileProp);
-            Properties properties = new Properties();
-            properties.load(is);
-
-            if (properties.getProperty(TOKEN_PROPERTY) != null) {
-                return properties.getProperty(TOKEN_PROPERTY);
-            }
-        } catch (Exception e) {
-            // intentionally ignored
-        } finally {
-            try {
-                if (is != null) {
-                    is.close();
-                }
-            } catch (IOException e) {
-                // intentionally ignored
+    //Properties
+    private static Properties properties = null;
+
+    public static Properties getProperties(String dirName, String fileName) {
+        if (properties == null) {
+            File dir = new File(dirName);
+            File fileProp = new File(dir, fileName);
+            try (InputStream is = new FileInputStream(fileProp)) {
+                Properties properties = new Properties();
+                properties.load(is);
+                if (properties.isEmpty())
+                    throw new RuntimeException("Properties file cannot be empty");
+                Utils.properties = properties;
```

Comment:
no need to prefix this with `Utils.` as you are in the class containing this field